src/pipe_log.rs

```diff
@@ -172,6 +173,9 @@ pub trait PipeLog: Sized {
     /// Reads some bytes from the specified position.
     fn read_bytes(&self, handle: FileBlockHandle) -> Result<Vec<u8>>;
 
+    /// Reads bytes from multi blocks using 'Async IO'.
+    fn async_read_bytes(&self, ents_idx: &mut Vec<EntryIndex>) -> Result<Vec<Vec<u8>>>;
```

Comment:
As mentioned before, pipe is not aware of the "entry" concept. Here should use `Vec<FileBlockHandle>`.